### PR TITLE
aws-sam-cli: 1.90.0 -> 1.103.0

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

https://github.com/aws/aws-sam-cli/releases/tag/v1.103.0
https://github.com/aws/aws-sam-cli/compare/v1.90.0...v1.103.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/boto3-stubs/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{ lib
```
Comment (tjni) [inline pkgs/development/python-modules/boto3-stubs/default.nix:1]: I think this already exists under the name `mypy-boto3-builder`.

**Comment by anthonyroussel** [comment]
Waiting for #253315 to continue

Since v1.90, aws-sam-cli requires the following dependencies:

* boto3-stubs-apigateway
* boto3-stubs-cloudformation
* boto3-stubs-ecr
* boto3-stubs-iam
* boto3-stubs-lambda
* boto3-stubs-s3
* boto3-stubs-schemas
* boto3-stubs-secretsmanager
* boto3-stubs-signer
* boto3-stubs-stepfunctions
* boto3-stubs-sts
* boto3-stubs-xray

See https://github.com/aws/aws-sam-cli/blob/develop/requirements/base.txt#L35

**Comment by mbalatsko** [comment]
I'm preparing PR, in which all missing mypy-boto3 packages will be added. I have created an automated script for that, so that should be today ready. I will let you now, once PR is up

**Comment by mbalatsko** [comment]
Pr was created: https://github.com/NixOS/nixpkgs/pull/258679

**Comment by fabaff** [comment]
> Added python310Packages.boto3-stubs which is now required by aws-sam-cli

`python310Packages.boto3-stubs` is also part of https://github.com/NixOS/nixpkgs/pull/258679

**Comment by anthonyroussel** [comment]
@ofborg build aws-sam-cli python311Packages.aws-sam-translator

**Comment by anthonyroussel** [comment]
> I'm preparing PR, in which all missing mypy-boto3 packages will be added. I have created an automated script for that, so that should be today ready. I will let you now, once PR is up

@mbalatsko @fabaff Thanks for your help.
I have updated my PR with the new boto3-stubs dependency packages.

**Comment by anthonyroussel** [comment]
Updated to 1.103.0

**Comment by marsam** [comment]
It looks like some tests are failing on Ofborg

**Pull request merged**